### PR TITLE
NEUBURG-87: disable confirmation link after48 hours

### PR DESCRIPTION
**Pull request by michael-markl**

### CODE REVIEW

File: src/api/offers.js [hunk 1]
```diff
@@ -88,6 +90,9 @@ export default ({offerService, errorService}: { offerService: OfferService, erro
       } else if (offer.expirationDate <= Date.now()) {
         const errorResponse = errorService.createOfferExpiredErrorResponse(token)
         response.status(HttpStatus.GONE).json(errorResponse)
+      } else if (offer.confirmed === false && offer.createdDate < Date.now() - CONFIRMATION_PERIOD) {
```
Comment (NKlug) [inline src/api/offers.js]: Why do you check for `=== false`? Should `!offer.confirmed` be sufficient?

File: src/api/offers.js [hunk 2]
```diff
@@ -88,6 +90,9 @@ export default ({offerService, errorService}: { offerService: OfferService, erro
...
+        const errorResponse = errorService.createConfirmationLinkExpiredErrorResponse(token)
+        response.status(HttpStatus.GONE).json(errorResponse)
```
Comment (maxammann) [inline src/api/offers.js]: I think we should hide here the fact that the offer is still in our system. If the cleanup-job hasn't run yet then it returns GONE. As soon as it run it returns NOT_FOUND. I think this is inconsistent.

File: src/api/offers.js [hunk 3]
```diff
@@ -12,6 +12,8 @@ import ErrorService from '../services/ErrorService'
 
 const develop = process.env.NODE_ENV === 'development'
 
+const CONFIRMATION_PERIOD = 1000 * 60 * 60 * 48
```
Comment (maxammann) [inline src/api/offers.js]: In the context of eslint these are still magic values:

https://github.com/Integreat/neuburg-backend/pull/28/files#diff-b528c50833f673d233fa7dc20b02600eR18

**Review by maxammann — CHANGES_REQUESTED** [review]

File: src/api/offers.js (hunk 2 above)
Comment (michael-markl) [reply src/api/offers.js]: Hmm... you're probably right...

File: src/api/offers.js (hunk 3 above)
Comment (michael-markl) [reply src/api/offers.js]: Done

File: src/api/offers.js (hunk 2 above)
Comment (michael-markl) [reply src/api/offers.js]: Done

File: src/api/offers.js (hunk 1 above)
Comment (michael-markl) [reply src/api/offers.js]: Done

**Review by steffenkleinle — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

**Review by NKlug — APPROVED** [review]

**Pull request merged**